### ISSUE TITLE
Using type() to compare types - cannot use class that has `str` as base class

### ISSUE THREAD

**Issue by netrix** [aio-libs/aioredis-py#856]
It's not possible to use `StrEnum(str, Enum)` object as a key because of following code: https://github.com/aio-libs/aioredis/blob/231d0418988fd870c3d26d60eec65e2fc13d0144/aioredis/util.py#L35

Unfortunately this code requires argument to by strictly `str` type instead of being a subclass of `str`.

What I'd like to achieve is following:

```python
from enum import Enum

class StrEnum(str, Enum):
     AAA = "aaa"

# ....

rdb.hmget("key", StrEnum.AAA)
```

**Comment by seandstewart** [comment]
@netrix - Thanks for the submission!

I've implemented something similar here:

https://github.com/seandstewart/typical/blob/47a937fa1586c2b5bc0be6a6f89d06b089df3f3d/typic/util.py#L515-L535

If you have a moment to provide an implementation, I'd love to see a PR for this!

**Comment by seandstewart** [comment]
This issue should be resolved as of #891 - please feel free to pull in the latest master and test to ensure this is the case.
